mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/dialect/impl/MongoDBAssociationSnapshot.java

```diff
@@ -46,11 +47,11 @@ public DBObject getQueryObject() {
 
 		if ( storageStrategy == AssociationStorageStrategy.IN_ENTITY ) {
 			if ( associationKey.getMetadata().isOneToOne() ) {
-				Object oneToOneValue = getValueOrNull( document, associationKey.getMetadata().getCollectionRole(), Object.class );
+				Object oneToOneValue = getValueOrNull( (Map<?,?>) document, associationKey.getMetadata().getCollectionRole(), Object.class );
```

Comment:
`DBObject` is not safe to be cast to `Map`. The default implementation `BasicDBObject` indeed is, but it's a slippery slope we'd enter when relying on it. I'd prefer to duplicate that code for now (apparently, MongoDB driver 3 comes with a replacement for `DBObject` et al. anyways, so we may re-visit the topic then).